bin/mail/2017-08-30-pd-pre-workshop-survey-correction/generate-list

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env ruby
+require_relative '../mailing-common/mailing-list-utils'
+require_relative '../../../dashboard/config/environment'
+
+# The original mail with the erroneous link was sent as the 10-day reminder email on the morning of 2017-08-30
+# Make sure we're running an equivalent query of 10 days before 8-30
+days_before = (Date.parse('2017-08-30') + 10.days - Date.today).to_i
+raise 'Must be run before 2017-08-30' if days_before < 1
+
+workshops = Pd::Workshop.scheduled_start_in_days(days_before).
```

Comment:
Orthogonal, but maybe we should denormalize the db and have workshop start date as a field - then index the field for faster lookups.

Reply:
That's not a bad idea. We'll have to be careful that updating the sessions updates the parent workshop, but that shouldn't be too bad with ActiveRecord callbacks.